pymc_marketing/mmm/models/components/lagging.py

```diff
@@ -0,0 +1,313 @@
+from typing import Dict, Optional, Union
+
+import pymc as pm
+
+from pymc_marketing.mmm.transformers import geometric_adstock, weibull_adstock
+from pymc_marketing.mmm.utils import _get_distribution, _validate_model_config
+
+
+class BaseFunction:
+    def __init__(self, max_lagging, model):
+        self.max_lagging = max_lagging
+        self.model = model
+
+    def apply(self, data):
+        raise NotImplementedError("Subclasses must implement this method.")
```

Comment:
Think all of the defaults should go here rather than in an ifelse block that is hidden within a method. With defaults here, a user would be able to make their own saturation function that we don't have without having to change the ifelse block